tutorials/composite_bo_with_hogp.ipynb

```diff
@@ -0,0 +1,646 @@
+      "I0712 082633.467 <ipython-input-11-c3f919c3cba7>:85] Wall time: 17.381221\n",
+      "I0712 082633.469 <ipython-input-11-c3f919c3cba7>:22] rnd: -0.126It 13/15, best obs.: , ei: -0.005It 13/15, best obs.: , ei_hogp_cf: -0.011\n",
+      "I0712 082652.786 <ipython-input-11-c3f919c3cba7>:85] Wall time: 19.317743\n",
+      "I0712 082652.788 <ipython-input-11-c3f919c3cba7>:22] rnd: -0.126It 14/15, best obs.: , ei: -0.005It 14/15, best obs.: , ei_hogp_cf: -0.011\n",
+      "I0712 082701.116 <ipython-input-11-c3f919c3cba7>:85] Wall time: 8.329369\n",
+      "I0712 082701.118 <ipython-input-11-c3f919c3cba7>:22] rnd: -0.126It 15/15, best obs.: , ei: -0.005It 15/15, best obs.: , ei_hogp_cf: -0.011\n",
+      "I0712 082708.962 <ipython-input-11-c3f919c3cba7>:85] Wall time: 7.844674\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Finished with trial:  3\n"
+     ]
+    }
+   ],
+   "source": [
+    "all_objective_vals = []\n",
+    "\n",
+    "with gpt_settings.cholesky_jitter(1e-4):\n",
+    "    for trial in range(n_trials):\n",
+    "        print(\"Beginning with trial: \", trial+1)\n",
+    "        c_batched, objective, bounds, num_samples = prepare_data(device=device, dtype=dtype)\n",
+    "\n",
+    "        train_X_init = gen_rand_points(bounds, n_init)\n",
+    "        train_Y_init = c_batched(train_X_init)\n",
+    "\n",
+    "        # these will keep track of the points explored\n",
+    "        train_X = {k: train_X_init.clone() for k in models_used}\n",
+    "        train_Y = {k: train_Y_init.clone() for k in train_X}\n",
+    "\n",
+    "        # run the BO loop\n",
+    "        for i in range(n_batches):\n",
+    "            tic = time.time()\n",
+    "\n",
+    "            # get best observations, log status\n",
+    "            best_f = {k: objective(v).max().detach() for k, v in train_Y.items()}\n",
+    "\n",
+    "            logging.info(\n",
+    "                f\"It {i+1:>2}/{n_batches}, best obs.: \"\n",
+    "                \", \".join([f\"{k}: {v:.3f}\" for k, v in best_f.items()])\n",
+    "            )\n",
+    "\n",
+    "            # generate random candidates\n",
+    "            cands = {}\n",
+    "            cands[\"rnd\"] = gen_rand_points(bounds, batch_size)\n",
+    "\n",
+    "            optimize_acqf_kwargs = {\n",
+    "                \"q\": batch_size,\n",
+    "                \"num_restarts\": 10,\n",
+    "                \"raw_samples\": 512,\n",
+    "            }\n",
+    "            sampler = IIDNormalSampler(num_samples=128)\n",
+    "\n",
+    "            train_Y_ei = objective(train_Y[\"ei\"]).unsqueeze(-1)\n",
+    "            model_ei = SingleTaskGP(\n",
+    "                train_X[\"ei\"],\n",
+    "                train_Y_ei,\n",
+    "                input_transform=Normalize(train_X[\"ei\"].shape[-1]),\n",
+    "                outcome_transform=Standardize(train_Y_ei.shape[-1]),\n",
+    "            )\n",
```

Comment:
Should explain before this cell that you'll do the comparison against BO on the aggregate metrics and random.